core/src/test/java/com/radixdlt/api/core/NetworkSubmitTransactionTest.java

```diff
@@ -118,9 +118,93 @@ public void test_core_api_can_submit_and_commit_transaction() throws Exception {
     }
   }
 
+  @Test
+  public void test_transaction_rejected_when_same_payload_previously_committed() throws Exception {
+    try (var test = buildRunningServerTest()) {
+      test.suppressUnusedWarning();
+
+      var transaction = TransactionBuilder.forTests().prepare();
+
+      // Submit transaction
+      getTransactionApi()
+          .transactionSubmitPost(
+              new TransactionSubmitRequest()
+                  .network(networkLogicalName)
+                  .notarizedTransactionHex(transaction.hexPayloadBytes()));
+
+      // Ensure it is committed
+      test.runUntilState(allCommittedTransactionSuccess(transaction.raw()), 1000);
+
+      // Submit the same transaction again
+      var errorResponse =
+          assertErrorResponseOfType(
+              () ->
+                  getTransactionApi()
+                      .transactionSubmitPost(
+                          new TransactionSubmitRequest()
+                              .network(networkLogicalName)
+                              .forceRecalculate(true)
+                              .notarizedTransactionHex(transaction.hexPayloadBytes())),
+              TransactionSubmitErrorResponse.class);
+
+      assertThat(errorResponse.getCode()).isEqualTo(400);
+      var details = (TransactionSubmitRejectedErrorDetails) errorResponse.getDetails();
+      assertThat(details.getIsIntentRejectionPermanent()).isTrue();
+      assertThat(details.getIsPayloadRejectionPermanent()).isTrue();
+      assertThat(details.getIsRejectedBecauseIntentAlreadyCommitted()).isTrue();
```

Comment:
_(note from author)_

I added these tests so that we precisely see how the 2 flavors differ (same intent in different payloads vs same payload) in terms of returned booleans.

E.g. my personal doubt is in this very line:
The test is named `rejected_when_same_payload`.
Logically the Node can reject it because same payload is committed (and this is a more "narrow" reason than "same intent committed").
So should we return `is_rejected_because_intent_already_committed` true or rather false?
(apparently I have problem with the word "because" :D)

Reply:
Technically, it is rejected because of `intent` (and not the `payload`) :P.

Reply:
hah, I see, so the official interpretation is:

"Yeah it would be totally legal for 2 committed transactions to have the same payload hash, but unfortunately it is not possible because that would mean the same intent hashes (which happens to be illegal)."

and then, indeed `IsRejectedBecauseIntentAlreadyCommitted == true` is valid for both cases 👍